packages/components/organisms/f-registration/test/specs/component/f-registration.component.spec.js

```diff
@@ -17,22 +29,71 @@ describe('f-registration component tests', () => {
         };
 
         // Act
-        RegistrationComponent.submitRegistrationForm(userInfo);
+        registration.submitForm(userInfo);
+        
+        // Assert
+        expect(registration.countOfErrorsDisplayed()).toEqual(4)
+        expect(registration.isEmptyErrorDisplayed(field)).toBe(true);
+    });
+
+    forEach(['firstName', 'lastName', 'email'])
+    .it('should display error when input is invalid', field => {
+        // Arrange
+        const userInfo = {
+            firstName: '123*',
+            lastName: '456*',
+            email: '***@**', 
+            password: 'llanfairpwllgwyngyllgogerychwyr'
+        };
+
+        // Act
+        registration.submitForm(userInfo);
+        
+        // Assert
+        expect(registration.isInvalidErrorDisplayed(field)).toBe(true);
+    });
 
+    forEach(['firstName', 'lastName'])
+    .it('should display error when input is too long', field => {
+        // Arrange
+        const userInfo = {
+            firstName: 'abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghij',
+            lastName: 'abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghij',
+            email: 'ashton.adamms+jetest@just-eat.com', 
+            password: 'llanfairpwllgwyngyllgogerychwyr'
+        };
+
+        // Act
+        registration.submitForm(userInfo);
+        
         // Assert
-        expect(RegistrationComponent.isFirstNameEmptyErrorDisplayed()).toBe(true);
-        expect(RegistrationComponent.isLastNameEmptyErrorDisplayed()).toBe(true);
-        expect(RegistrationComponent.isEmailEmptyErrorDisplayed()).toBe(true);
-        expect(RegistrationComponent.isPasswordEmptyErrorDisplayed()).toBe(true);
+        expect(registration.isMaxLengthErrorDisplayed(field)).toBe(true);
     });
 
-    it.skip('should show and be able to click the legal documentation', () => {
+    forEach(['firstName', 'lastName', 'email', 'password'])
```

Comment:
This test can be deleted IMO. We should only test stuff like this in CoreWeb where we're able to mock the underlying API's. Happy to explain on a call if this reasoning is still unclear :)

Reply:
Yeah I wondered this at the time actually - wasn't sure how to go about showing that no field errors are displayed when the form is filled out correctly. We have some similar tests in f-checkout but can't assert on them until there's a redirect anyway so guessing those could go too? Waiting for the API makes more sense so happy to delete!